platform/golden/scripts/serial.sh

```diff
@@ -0,0 +1,24 @@
+#!/bin/sh
+
+cat > /tmp/fixes.sh <<EOF
+#!/bin/bash
+set -x
+echo 'GRUB_CMDLINE_LINUX="console=ttyS0"' >> /etc/default/grub
+update-grub
+
+if [ -f /etc/udev/rules.d/70-persistent-net.rules ] ; then 
+  rm /etc/udev/rules.d/70-persistent-net.rules
+  ln -s /dev/null /etc/udev/rules.d/70-persistent-net.rules
+fi
+
+echo 'packer ALL=(ALL) NOPASSWD:ALL'>> /etc/sudoers.d/91-packer
+
+passwd -d packer
+
+mkdir /home/packer/.ssh
+chmod 700 /home/packer/.ssh
+curl -sLo - https://github.com/sdake.keys >> /home/packer/.ssh/authorized_keys
+curl -sLo - https://github.com/rstarmer.keys >> /home/packer/.ssh/authorized_keys
```

Comment:
I am not really sure how we should handle dynamic secret management. For now, I have started a [wiki entry](https://github.com/artificialwisdomai/origin/wiki/Secrets) so we know where our secrets are stored or consumed in our repo.

Reply:
I think we can enable this as a provisioning step.  If we're goign to use Ansible, then it would likely be easier to add an ansible provisioner that loads the right keys at build time, or even better at image deploy time so that there is nothing static in the base/golden image.

Reply:
Right. I think some things should be dynamic, and the rest should be baked (with packer). Dynamically loaded credentials make sense, far more sense then static.

But, I spent like 5+ years working with people at Red Hat that battled and battled over this credential issue. My recommendation - tech debt it for now, we will sort out the workflow later.

Another way to introduce SSH credentials is via the kernel command line.